examples/3d_rendering_processing/transform_feedback.lpr

```diff
@@ -0,0 +1,135 @@
+{
+  Copyright 2021 Trung Le (Kagamma).
+
+  This file is part of "Castle Game Engine".
+
+  "Castle Game Engine" is free software; see the file COPYING.txt,
+  included in this distribution, for details about the copyright.
+
+  "Castle Game Engine" is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+
+  ----------------------------------------------------------------------------
+}
+
+{ Example how to use TGLSLProgram.SetTransformFeedbackVaryings }
+
+program transform_feedback;
+
+{$macro on}
+{$define nl:=+ LineEnding +}
+
+uses
+  GL, GLExt,
+  CastleVectors, X3DNodes, CastleWindow, CastleLog,
+  CastleUtils, SysUtils, CastleApplicationProperties,
+  CastleViewport, CastleTimeUtils, CastleGLShaders;
+
+const
+  VertexArray: packed array[0..2] of TVector2 = (
+    (Data: (-1, 0)), (Data: (1, 0)), (Data: (0, 1))
+  );
+  TransformFeedbackVertexShaderSource: String =
+'#version 330'nl
+'layout(location = 0) in vec2 inVertex;'nl
+'out vec2 outVertex;'nl
+'float atan2(vec2 v) {'nl
+'  return v.x == 0.0 ? sign(v.y) * 3.1415 / 2.0 : atan(v.y, v.x);'nl
+'}'nl
+'void main() {'nl
+'  float a = atan2(inVertex) + 0.01;'nl
+'  outVertex = vec2(cos(a), sin(a));'nl
+'}';
+
+  RenderVertexShaderSource: String =
+'#version 330'nl
+'layout(location = 0) in vec2 inVertex;'nl
+'void main() {'nl
+'  gl_Position = vec4(inVertex, 0.0, 1.0);'nl
+'}';
+
+  RenderFragmentShaderSource: String =
+'#version 330'nl
+'out vec4 outColor;'nl
+'void main() {'nl
+'  outColor = vec4(1.0);'nl
+'}';
+
+var
+  Window: TCastleWindowBase;
+  Viewport: TCastleViewport;
+  VAOs,
+  VBOs: array[0..1] of GLint;
```

Comment:
We usually place variable declarations of the same type on the same line in CGE, so make it

```
VAOs, VBOs: array[0..1] of TGLint;
```

We usually write declarations like that. In some exceptions we make it multi-line but then we indent the 2nd line, like this:

```pascal
  SomeVariable, SomeLongLongVariableName,
    SomeReallyReallyLongVairiableName: TSomething;
```

This way it is easier to "parse" the variable declarations by a human eye. I fully know that it is only a matter of taste :) -- so in this case it's just better to follow existing conventions of CGE.

Also use `Txxx` types, so `GLint` -> `TGLint`.